umbral/umbral.py

```diff
@@ -358,13 +358,13 @@ def check_challenge(self, capsule, cFrag, challenge_resp, pub_a, pub_b):
         z2 = challenge_resp.bn_kfrag_sig2
         z3 = challenge_resp.bn_sig
 
-        g_y = (g * z2) + (pub_a * z1)
```

Comment:
So it looks like you've set `__rmul__` to `__mul__` and then reversed the order of a lot of these operations.  Is this just so that the order of operations matches the language of the paper?  Or is there a deeper / darker purpose?  :-)

Reply:
There are two possible *standard* conventions to represent operations between elements of algebraic groups used in cryptography:
- Multiplicative notation: This notation assumes that the operation between elements of the group is the multiplication (e.g., `g * h`), and therefore, the operation between a scalar and an element of the group is the exponentiation (in code: `g ** n`), which mathematically is equivalent to `g * g * (n times) ...`. This is the notation used when using "generic groups", which means that the scheme is not using concrete information about the underlying mathematical structure of the group (e.g., if using elliptic curves, the scheme doesn't use things like the coordinates). This notation is used in the paper, and in practically all PRE schemes out there. 
- Additive notation: It assumes that the operation is the addition (e.g., `g + h`), and therefore, the operation between a scalar and an element of the group is the multiplication (in code: `n * g`), which mathematically is equivalent to `g + g + (n times) ... `. This notation is usually used in Elliptic Curve crypto, for historic reasons. Note that the scalar multiplication is usually represented with the scalar in the left when the group is commutative. 

The problem was that the operations of the Point class followed the additive notation (which is the tradition in ECC and in libraries like OpenSSL) but our scalar multiplication had the scalar in the right. tuxxy's changes simply accommodated the code to the additive notation.